### PR TITLE
feat(codepipeline): add construct for registering custom Actions

### LINKED ISSUE #17039: (aws-codepipeline-actions): custom-action-registration and common are not exported in index.ts
### What is the problem?

When you want to create your own custom action, you need to register it. Using `CustomActionRegistration` from `custom-action-registration.ts` makes this easy, and it's already exported from that ts file. However, to use it as part of the `@aws-cdk/aws-codepipeline-action` module, that class needs to be exported on the `index.ts` file in that directory. Currently, the contents of `custom-action-registration.ts` are not exported from the `index.ts`. This is also true for `sourceArtifactBounds` and `deployArtifactBounds` from the `common.ts` file.

### Reproduction Steps

```
import { CustomActionRegistration } from '@aws-cdk/aws-codepipeline-actions';
```

Module '"@aws-cdk/aws-codepipeline-actions"' has no exported member 'CustomActionRegistration'.
Module '"@aws-cdk/aws-codepipeline-actions"' has no exported member 'sourceArtifactBounds'.
Module '"@aws-cdk/aws-codepipeline-actions"' has no exported member 'deployArtifactBounds'.

### What did you expect to happen?

Expected the content of `custom-action-registration.ts` and `common.ts` in the `@aws-cdk/aws-codepipeline-actions` module to be available in the library module.

### What actually happened?

The content of `custom-action-registration.ts` and `common.ts` in the `@aws-cdk/aws-codepipeline-actions` module are not in the library module because they are not exported in the `index.ts`

### CDK CLI Version

1.128.0

### Framework Version

_No response_

### Node.js Version

v16.11.1

### OS

os x

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by rayjanwilson**
fixes #17039

all it does is add `./common` and `./custom-action-registration` to the modules `index.ts`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rayjanwilson** [comment]
> We need to make all of the properties of `CustomActionProperty` `readonly`. While we're in the area, we'll probably have to make a few more adjustments before we make this a public part of the API of the module (and hence we'll have to maintain it going forwards).
> 
> Are you up for that @rayjanwilson? 🙂

yep happy to give it a go and iterate towards what it should be. first time doing this =)

**Comment by skinny85** [comment]
@rayjanwilson Now you need to do the same for `CustomActionRegistrationProps` 😜.

```
@aws-cdk/aws-codepipeline-actions: [96mlib/custom-action-registration.ts[0m:[93m98[0m:[93m3[0m - [91merror[0m[90m JSII3008: [0mThe "executionUrl" property of struct "@aws-cdk/aws-codepipeline-actions.CustomActionRegistrationProps" must be "readonly". Rename "@aws-cdk/aws-codepipeline-actions.CustomActionRegistrationProps" to "ICustomActionRegistrationProps" if it is meant to be a behavioral interface.
```

**Comment by skinny85** [comment]
Still some errors:

```
@aws-cdk/aws-codepipeline-actions: error: [awslint:construct-ctor:@aws-cdk/aws-codepipeline-actions.CustomActionRegistration.<initializer>.params[0]] signature of all construct constructors should be "scope, id, props".   (expected="scope",actual="parent")
@aws-cdk/aws-codepipeline-actions: error: [awslint:props-default-doc:@aws-cdk/aws-codepipeline-actions.CustomActionRegistrationProps.entityUrl] Optional property must have @default documentation 
@aws-cdk/aws-codepipeline-actions: error: [awslint:props-default-doc:@aws-cdk/aws-codepipeline-actions.CustomActionRegistrationProps.executionUrl] Optional property must have @default documentation 
@aws-cdk/aws-codepipeline-actions: Error: /codebuild/output/src259442693/src/github.com/aws/aws-cdk/tools/@aws-cdk/cdk-build-tools/bin/cdk-awslint exited with error code 1
```

@rayjanwilson did you go through our ["Contributing" guide](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)? You should be able to see these errors when `build`ing locally, no need to wait for the PR build 🙂.

File: packages/@aws-cdk/aws-codepipeline/lib/common.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import { ActionArtifactBounds } from './action';
 
 /**
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/lib/common.ts:3]: I thought we resolved this already? 😃

Do not touch this file - leave it where it is, don't move it to the `@aws-cdk/aws-codepipeline` module, and do not make it public.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the effort @rayjanwilson, but, like I mentioned before, let's leave `common.ts` exactly where it is now, unchanged.

File: packages/@aws-cdk/aws-codepipeline/lib/common.ts (hunk 1 above)
Comment (rayjanwilson) [reply packages/@aws-cdk/aws-codepipeline/lib/common.ts]: we had, my bad. it's what i get for picking back up after a week or so. apologies. reverted

File: packages/@aws-cdk/aws-codepipeline/README.md [hunk 2]
```diff
@@ -109,6 +109,40 @@ or you can use the `IStage.addAction()` method to mutate an existing Stage:
 sourceStage.addAction(someAction);
 ```
 
+## Custom Action Registration
+
+To make your own custom CodePipeline Action requires registering the action provider. Look to the `JenkinsProvider` in `@aws-cdk/aws-codepipeline-actions` for an implementation example.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/README.md:114]: Be wary of long lines in this file:

```suggestion
To make your own custom CodePipeline Action requires registering the action provider.
Look to the `JenkinsProvider` in `@aws-cdk/aws-codepipeline-actions` for an implementation example.
```

**Review by skinny85 — APPROVED** [review]
Looks great, thanks @rayjanwilson!

**Pull request merged**